imu_sensor_broadcaster/include/imu_sensor_broadcaster/imu_transform.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 AIT Austrian Institute of Technology GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * Authors: Christoph Froehlich
+ * Adapted from
+ * https://github.com/ros-perception/imu_pipeline/blob/jazzy/imu_transformer/include/imu_transformer/tf2_sensor_msgs.h
+ */
+
+#ifndef IMU_SENSOR_BROADCASTER__IMU_TRANSFORM_HPP_
+#define IMU_SENSOR_BROADCASTER__IMU_TRANSFORM_HPP_
+
+#include <Eigen/Dense>
+#include "sensor_msgs/msg/imu.hpp"
+
+namespace imu_sensor_broadcaster
+{
+/**
+ * @brief Euler to quaternion Z-Y'-X'' convention
+ * https://eigen.tuxfamily.org/dox-devel/group__TutorialGeometry.html
+ */
+Eigen::Quaternion<double> quat_from_euler(double roll, double pitch, double yaw)
+{
+  return Eigen::Quaternion<double>(
+    Eigen::AngleAxisd(yaw, Eigen::Vector3d::UnitZ()) *
+    Eigen::AngleAxisd(pitch, Eigen::Vector3d::UnitY()) *
+    Eigen::AngleAxisd(roll, Eigen::Vector3d::UnitX()));
+}
+
+/**
+ * @brief Transforms a covariance array from one frame to another
+ */
+inline void transformCovariance(
+  const std::array<double, 9> & in, std::array<double, 9> & out, Eigen::Quaternion<double> r)
```

Comment:
```suggestion
inline void transformCovariance(
  const std::array<double, 9> & in, const Eigen::Quaternion<double> & r, std::array<double, 9> & out)
```

Can we change the order and have `r` as const, so the first arguments are const and then comes the output. Does it make sense?

r is the offset or the transformation that we need to apply. Am i right?.

Reply:
As I mentioned, I basically copied it over from the other repo. I usually order it like (output, input) like memcpy for example, I can apply that.